src/shared/components/lollipopMutationPlot/LollipopMutationPlotWrapper.tsx

```diff
@@ -0,0 +1,508 @@
+import Response = request.Response;
+import {observer} from "mobx-react";
+import {computed, observable, action} from "mobx";
+import _ from "lodash";
+import {longestCommonStartingSubstring} from "../../lib/StringUtils";
+import {findFirstMostCommonElt} from "../../lib/findFirstMostCommonElt";
+import ReactDOM from "react-dom";
+import {Form, Button, FormGroup, InputGroup, ControlLabel, FormControl} from "react-bootstrap";
+import fileDownload from "react-file-download";
+import "./styles.scss";
+
+type LollipopMutationPlotWrapperProps = {
+    dataStore:IMobXApplicationDataStore<Mutation[]>;
+    entrezGeneId:number;
+    hugoGeneSymbol?:string;
+};
+
+const mutationTypePriority:{[canonicalMutationType:string]:number} = {
+    "missense": 1,
+    "inframe": 2,
+    "truncating": 4,
+    "nonsense": 6,
+    "nonstop": 7,
+    "nonstart": 8,
+    "frameshift": 4,
+    "frame_shift_del": 4,
+    "frame_shift_ins": 5,
+    "in_frame_ins": 3,
+    "in_frame_del": 2,
+    "splice_site": 9,
+    "fusion": 10,
+    "silent": 11,
+    "other": 11
+};
+
+function lollipopMutationTypeSort(typeA:CanonicalMutationType, typeB:CanonicalMutationType) {
+    const priorityA = mutationTypePriority[typeA];
+    const priorityB = mutationTypePriority[typeB];
+    if (priorityA < priorityB) {
+        return -1;
+    } else if (priorityA > priorityB) {
+        return 1;
+    } else {
+        return typeA.localeCompare(typeB);
+    }
+}
+
+const MISSENSE_COLOR = "#008000";
+const TRUNCATING_COLOR = "#000000";
+const INFRAME_COLOR = "#8B4513";
+const OTHER_COLOR = "#8B00C9";
+
+@observer
+export default class LollipopMutationPlotWrapper extends React.Component<LollipopMutationPlotWrapperProps, {}> {
+
+    @observable private showControls:boolean = false;
+    @observable private _yMaxInput:number;
+    @observable private legendShown:boolean = false;
+    private plot:LollipopMutationPlot;
+    private handlers:any;
+
+    private getLollipopColor(mutations:Mutation[]):string {
+        const sortedCanonicalMutationTypes:CanonicalMutationType[] = mutations.map(m=>getCanonicalMutationType(m.mutationType)).sort(lollipopMutationTypeSort);
+        const chosenCanonicalType:CanonicalMutationType|undefined = findFirstMostCommonElt(sortedCanonicalMutationTypes);
+        if (chosenCanonicalType) {
+            const lollipopType:LollipopMutationType = getLollipopMutationTypeFromCanonical(chosenCanonicalType);
+
+            switch (lollipopType) {
+                case "missense":
+                    return MISSENSE_COLOR;
+                case "truncating":
+                    return TRUNCATING_COLOR;
+                case "inframe":
+                    return INFRAME_COLOR;
+                default:
+                    return OTHER_COLOR;
+            }
+        } else {
+            return "#ff0000"; // we only get here if theres no mutations, which shouldnt happen. red to indicate an error
+        }
+    }
+
+    readonly swissProtId = remoteData({
```

Comment:
you are right, this is more modular than putting into the store. But, maybe later we add separate cache for each data source in case we need to share the data.